exes/mqtt-broadcaster/src/main.cpp

```diff
@@ -0,0 +1,43 @@
+#include <async_mqtt/all.hpp>
+#include <boost/program_options.hpp>
+#include <tfc/ipc.hpp>
+#include "mqtt_broadcaster.hpp"
+
+auto main(int argc, char* argv[]) -> int {
+  try {
+    auto program_description{ tfc::base::default_description() };
+
+    std::string mqtt_host;
+    std::string mqtt_port;
+    std::string mqtt_username;
+    std::string mqtt_password;
+
+    program_description.add_options()("mqtt_host", boost::program_options::value<std::string>(&mqtt_host)->required(),
+                                      "ip address of mqtt broker")(
+        "mqtt_port", boost::program_options::value<std::string>(&mqtt_port)->required(), "port of mqtt broker")(
+        "mqtt_username", boost::program_options::value<std::string>(&mqtt_username), "username of mqtt broker")(
+        "mqtt_password", boost::program_options::value<std::string>(&mqtt_password), "password of mqtt broker");
+
+    tfc::base::init(argc, argv, program_description);
+
+    asio::io_context ctx{};
+
+    tfc::ipc_ruler::ipc_manager_client ipc_client{ ctx };
+
+    const std::shared_ptr<async_mqtt::endpoint<async_mqtt::role::client, async_mqtt::protocol::mqtt>> mqtt_client =
+        std::make_shared<async_mqtt::endpoint<async_mqtt::role::client, async_mqtt::protocol::mqtt>>(
+            async_mqtt::protocol_version::v5, ctx.get_executor());
```

Comment:
Can `ipc_client` and `mqtt_client` be constructed within mqtt_broadcaster?

Reply:
I can move mqtt_client but the ipc_client is needed for testing.

Reply:
you can use different constructor for testing

Reply:
you are right, moving the construction of mqtt_client and ipc_client to the normal constructor instead of being defined before.